### DISCUSSION TITLE
Gatsby Styling not being applied on initial render

### DISCUSSION THREAD

**Discussion by avi312singh** [gatsbyjs/gatsby#32149] [Help] [answered]
## Description

When my app is initially rendered the correct styling is not rendered with the pages(rendered as children from navDrawer), this only occurs during gatsby build and gatsby develop works fine.

e.g. 

Below code from parent component (navDrawer.jsx) rendering pages as children
```
const { darkMode } = useDarkThemeContext()
```

```
  const linkStyles = {
    textDecoration: "none",
    color: darkMode ? "white" : "black",
  }
```

```
[...].map((link, index) => (
          <ListItem key={link.text}>
            <ListItemIcon>{link.icon}</ListItemIcon>
            <Link
              to={link.href}
              activeStyle={activeStyle}
              style={linkStyles}
            >
              {link.text}
            </Link>
          </ListItem>
        ))}
```
**On first render** on any page during local develop and gatsby buld && gatsby serve both return true for dark mode
`console.log('darkMode ', darkMode)`
`true`

I tried to use the dev_ssr flag inside config but that did not seem to help fix this prod bug

### Steps to reproduce

Navigate to site -> [**Website production link**](https://falltoyourdeath.gtsb.io/)
Click on another nav link and the correct styling will appear, however refresh the page again and the styling will go.

Website code: [**Repository Link**](https://github.com/avi312singh/FTYD-Fe/blob/main/src/components/NavDrawer/NavDrawer.jsx#L117)

### Expected result

After refreshing the page the styling should remain the same as it does in the local environment (**gatsby develop**) - however this is not the same with **gatsby build && gatsby serve**

### Actual result

Styling disappears after refreshing the page and causing website to make initial render again

### Environment

  ```
System:
    OS: macOS 11.4
    CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.0.0 - ~/.nvm/versions/node/v14.0.0/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 6.14.4 - ~/.nvm/versions/node/v14.0.0/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 91.0.4472.114
    Edge: 91.0.864.59
    Firefox: 80.0.1
    Safari: 14.1.1
  npmPackages:
    gatsby: ^3.7.2 => 3.7.2
    gatsby-plugin-image: ^1.8.0 => 1.8.0
    gatsby-plugin-material-ui: ^3.0.1 => 3.0.1
    gatsby-plugin-react-helmet: ^4.7.1 => 4.7.1
    gatsby-plugin-sharp: ^3.8.0 => 3.8.0
    gatsby-source-filesystem: ^3.8.0 => 3.8.0
    gatsby-source-graphql: ^3.0.1 => 3.7.1
  npmGlobalPackages:
    gatsby-cli: 3.7.1
```

**Comment by avi312singh** [comment]
Not sure why this was closed? To be clear there is a different result after a gatsby build and running it locally using gatsby develop, it seems to be a bug with gatsby

**Reply by hustcc** [reply]
+1

**Comment by LekoArts** [comment, marked as answer]
You're running into an error with Rehydration which happens if your build output (e.g. the HTML) is different from what then the client loads. This is not something specific to Gatsby and can happen in any application that uses Rehydration. You can find other discussions that have solved this: https://github.com/gatsbyjs/gatsby/discussions?discussions_q=rehydration

**Reply by avi312singh** [reply]
<3
